app/controllers/api/base_controller/generic.rb

```diff
@@ -61,6 +61,22 @@ def delete_resource(type, id = nil, _data = nil)
         delete_resource_action(klass, type, id)
       end
 
+      def request_retire_resource(type, id, _data = nil)
+        if api_user_role_allows?('miq_request_approval')
+          klass = collection_class(type)
+          if id
+            msg = "Retiring #{type} id #{id} immediately as a request."
+            resource = resource_search(id, type, klass)
+            api_log_info(msg)
+            klass.make_retire_request(resource.id)
+          else
+            raise BadRequestError, "Must specify an id for retiring a #{type} resource"
+          end
+        else
+          raise ForbiddenError, "User lacking correct permissions to approve a #{type} retire as a request."
```

Comment:
can you add a test for this bad path, i.e. authorize user with service_retire and not miq_request_approval ?